subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c

```diff
@@ -232,6 +263,23 @@ int lll_deinit(void)
 		irq_disable(HAL_SWI_JOB_IRQ);
 	}
 
+	/* Disconnect dynamic ISRs used */
+#if defined(CONFIG_BT_CTLR_DYNAMIC_INTERRUPTS)
+#if defined(CONFIG_DYNAMIC_DIRECT_INTERRUPTS)
+	irq_disconnect_dynamic(RADIO_IRQn, CONFIG_BT_CTLR_LLL_PRIO,
```

Comment:
If no shared interrupts are enabled disabling it should be enough. The implementation of z_isr_install is very simple and it is just a copy of param and rundine into the right place in the array (see zephyr/arch/common/dynamic_isr.c file). So there is no sense of calling disconnect if no shared interrupts are enabled.

Reply:
Updated.